daemon/api.go

```diff
@@ -254,46 +254,67 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 	}
 	defer lock.Unlock()
 
-	sources := make([]string, 1, 2)
-	sources[0] = "local"
-	// we're not worried if the remote repos error out
-	found, _ := newRemoteRepo().All()
-	if len(found) > 0 {
-		sources = append(sources, "store")
+	results := make(map[string]map[string]interface{})
+	sources := make([]string, 0, 2)
+	query := r.URL.Query()
+
+	var includeStore, includeLocal bool
+	if len(query["sources"]) > 0 {
+		for _, v := range strings.Split(query["sources"][0], ",") {
+			if v == "store" {
+				includeStore = true
+			} else if v == "local" {
+				includeLocal = true
+			}
+		}
+	} else {
+		includeStore = true
+		includeLocal = true
 	}
 
-	sort.Sort(byQN(found))
+	var bags map[string]*lightweight.PartBag
 
-	bags := lightweight.AllPartBags()
+	if includeLocal {
+		sources = append(sources, "local")
+		bags = lightweight.AllPartBags()
 
-	results := make(map[string]map[string]interface{})
-	for _, part := range found {
-		name := part.Name()
-		origin := part.Origin()
+		for _, v := range bags {
+			m := v.Map(nil)
+			name, _ := m["name"].(string)
+			origin, _ := m["origin"].(string)
 
-		url, err := route.URL("name", name, "origin", origin)
-		if err != nil {
-			return InternalError("can't get route to details for %s.%s: %v", name, origin, err)
-		}
+			resource := "no resource URL for this resource"
+			url, _ := route.URL("name", name, "origin", origin)
```

Comment:
Good spot. This was the existing code but I'll update it to be consistent with below.